### ISSUE TITLE
Bug: list contributor search hitting `<CR>` doesn't do anything

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/app#2400]
### Describe the bug

Here's a sample video of what I'm seeing on the List edit / creation page:

https://github.com/open-sauced/app/assets/23109390/4cdbd3de-8502-47fc-bf04-dd79cd8fac08

You'll notice the drop down 'pops': that's when I'm hitting "enter" (or more agnostically, `<CR>`)

My expectation when hitting enter on a list would be that it returns search results for that query on a fuzzy basis. The only way to really make the returned list useful is to click on an entry (which you'll notice I do in the video above)

TLDR: hitting enter when searching should return and generate a list.

Noticing this on beta and prod

### Steps to reproduce

1. Create a List
2. Add some collaborators
3. Use the search bar with a query
4. Hit enter / `<CR>`

### Browsers

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by jpmcb** [comment]
Possibly also related to: https://github.com/open-sauced/app/issues/988 which seems to have similar problems with the search for repos

**Comment by kanugurajesh** [comment]
can i work on this issue ?

**Comment by jpmcb** [comment]
This _might_ need to be core team work: I don't have a good intuition on what's happening with the search under the hood or if it will require API changes
